test/utils/validateUserProperties.test.ts

```diff
@@ -0,0 +1,108 @@
+import isValidUsername from "utils/isValidUsername"
+
+describe("WHEN validating a users username", () => {
+  describe("AND a valid username is entered", () => {
+    it.each([["user.name"], ["user_name"], ["user-name"], ["username"]])(
+      "GIVEN %p it SHOULD return true",
+      (usernameToBeValidated) => {
+        // When
+        const result = isValidUsername(usernameToBeValidated)
+        // Then
+        expect(result).toBe(true)
+      }
+    )
+  })
+  describe("AND a non alphanumeric character is at the beginning of the proposed username", () => {
+    it.each([
+      [".username"],
+      ["/username"],
+      ["?username"],
+      ["_username"],
+      ["-username"],
+      ["~username"],
+      ["{username"],
+      ["}username"],
+      ["|username"],
+      ["\\username"],
+      ["[username"],
+      ["]username"]
+    ])("GIVEN %p it SHOULD return false", (usernameToBeValidated) => {
+      // When
+      const result = isValidUsername(usernameToBeValidated)
+      // Then
+      expect(result).toBe(false)
+    })
+  })
+  describe("AND a space is at the beginning of the proposed username", () => {
+    it("SHOULD return false", () => {
+      // Given
+      const proposedUsername = " username"
+      // When
+      const result = isValidUsername(proposedUsername)
+      // Then
+      expect(result).toBe(false)
+    })
+  })
+
+  describe("AND a space is included in the proposed username", () => {
+    it("SHOULD return false", () => {
+      // Given
+      const proposedUsername = " user name"
+      // When
+      const result = isValidUsername(proposedUsername)
+      // Then
+      expect(result).toBe(false)
+    })
+  })
+
+  describe("AND a space is at the end of the proposed username", () => {
+    it("SHOULD return false", () => {
+      // Given
+      const proposedUsername = " user name"
```

Comment:
the space is not at the end of the username

Reply:
whoops -_-

Reply:
Review comment 1